autoPyTorch/datasets/base_dataset.py

```diff
@@ -330,13 +330,19 @@ def get_dataset_for_training(self, split_id: int, train: bool, subset: int = 0)
         to provide training data to fit a pipeline
 
         Args:
-            split (int): The desired subset of the dataset to split and use
+            split_id (int): which split id to get from the splits
+            train (bool): whether the train or valid transforms are to be applied
+            subset (int, default=0): 0 is for train_indices, 1 is for valid_indices
```

Comment:
To me, `train != subset` looks always true.
I mean we do not need to take `subset`.
Or even if we need it, we should rename it such as `which_subset: Literal['train', 'val']`.

Reply:
I agree with your comment. I'll change it to just have train as the argument and use not train for the subset. I don't think we need a literal as it is a bit redundant to pass both train=True and subset=train